WordPress/gutenberg
# Remove deprecated `behaviors` syntax

## What?
Removes the deprecated `behaviors` syntax. 

Behaviors UI was originally introduced in https://github.com/WordPress/gutenberg/pull/49972. It was meant to be a way to add reusable pieces of interactivity to the frontend of sites. 

That implementation was removed in https://github.com/WordPress/gutenberg/pull/53851 and https://github.com/WordPress/gutenberg/pull/54509 and never merged into Core. Backwards-compatibility layer was added for users of the GB plugin.

## Why?
It should have already been deprecated in 17.0 because we only offer backwards-compatibility for 2 releases 🙂.

## How?
Removing 2 files and references to `behaviors` and `legacyLightboxSettings`.

## Testing Instructions

1. Open a post or page.
2. Insert an image block.
3. The "Expand on click" functionality should continue to work fine.
4. Try to insert the block that contains the removed syntax and it should NOT work anymore: 
   ```
   
   <figure class="wp-block-image size-large"><img src="http://gutenberg.local/wp-content/uploads/2023/11/alan-hardman-SU1LFoeEUkk-unsplash-1024x678.jpg" alt="" class="wp-image-157"/></figure>
   
   ```
